src/client/Microsoft.Identity.Client/AppConfig/PopAuthenticationConfiguration.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Microsoft.Identity.Client.AuthScheme.PoP;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// Configuration properties used to construct a Proof of possesion request
+    /// </summary>
+    public class PopAuthenticationConfiguration
+    {
+        /// <summary>
+        /// An HTTP uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request.
+        /// </summary>
+        public Uri RequestUri { get; set; }
+
+        /// <summary>
+        /// The HttP method to be bound to the request
+        /// </summary>
+        public HttpMethod PopHttpMethod { get; set; }
+
+        /// <summary>
+        /// A provider that can handle the asymmetric key operations needed by POP, that encapsulates a pair of public and private keys
+        /// and some typical crypto operations
+        /// </summary>
+        public IPoPCryptoProvider PopCryptoProvider { get; set; }
+
+        /// <summary>
+        /// This will be populated with the POP authentication header.
+        /// </summary>
+        public AuthenticationHeaderValue PopAuthenticationRequestHeader {get; private set;}
```

Comment:
I'm not sure what this is. If you're not sure either, delete it :)

Reply:
We used to populate the user provided HTTP Request headers with the authentication headers calculated form the key. Im assuming this is no longer needed?

This value was already being used for tests.

Reply:
It doesn't really make that much sense since this is a config option, but you cannot set this property. 

The way consumers are supposed to retrieve the pop token is via `AuthenticationResult`. We can provide more covenient mechasism upon demand.

Reply:
```csharp
 var r2 = await confidentialApp.AcquireTokenForClient(s_keyvaultScope)
              .WithProofOfPosession(popConfig)
              .ExecuteAsync(CancellationToken.None)
              .ConfigureAwait(false);
```

![image](https://user-images.githubusercontent.com/12273384/96627312-924b3300-1308-11eb-857e-b412e6c38368.png)